parsers/events.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+let addedCharacters = 0;
```

Comment:
Looks like you may have unintentionally left some global state? This is likely to have unintentional behaviour the second time this code runs. If my reading is right, you can exercise the bug by having two phrases in a verse with multiple chords over a single word. Let's add a test for that:

```
Am    Bm
Wonderful
Am Bm
Amazing
```

Reply:
Yup, definitely forgot about this one! I moved everything into a class to handle this state as well as added another wonderful test.